File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/ReadyStatusSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.GeolocationRuleset
import com.usercentrics.sdk.UsercentricsReadyStatus

internal fun UsercentricsReadyStatus.serialize(): Any {
    return mapOf(
        "shouldCollectConsent" to shouldCollectConsent,
        "consents" to consents.map { it.serialize() },
        "geolocationRuleset" to geolocationRuleset?.serialize(),
        "location" to location.serialize(),
    )
}

internal fun GeolocationRuleset.serialize(): Any {
    return mapOf(
        "activeSettingsId" to activeSettingsId,
        "bannerRequiredAtLocation" to bannerRequiredAtLocation,
    )
}

File: ios/Classes/Serializer/ReadyStatusSerializer.swift
import Usercentrics

extension UsercentricsReadyStatus {
    func serialize() -> Any {
        return [
            "shouldCollectConsent": self.shouldCollectConsent,
            "consents": self.consents.map { $0.serialize() },
            "geolocationRuleset": self.geolocationRuleset?.serialize(),
            "location": self.location.serialize()
        ]
    }
}


extension GeolocationRuleset {
    func serialize() -> Any {
        return [
            "activeSettingsId": self.activeSettingsId,
            "bannerRequiredAtLocation": self.bannerRequiredAtLocation
        ]
    }
}
